### PR TITLE
Update TimeSpan Parsing Test

### PR DESCRIPTION
**Pull request by tarekgh**
This PR depends on https://github.com/dotnet/coreclr/pull/21968

CC @joshfree @quole @krwq @danmosemsft

### CODE REVIEW

File: src/System.Runtime/tests/System/TimeSpanTests.cs [hunk 1]
```diff
@@ -538,7 +538,11 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
             yield return new object[] { "1:1:1.00001", CultureInfo.InvariantCulture, new TimeSpan(36610000100) };
             yield return new object[] { "1:1:1.000001", CultureInfo.InvariantCulture, new TimeSpan(36610000010) };
             yield return new object[] { "1:1:1.0000001", CultureInfo.InvariantCulture, new TimeSpan(36610000001) };
-            yield return new object[] { "1:1:1.00000001", CultureInfo.InvariantCulture, new TimeSpan(36610000001) };
+
+            if (PlatformDetection.IsFullFramework)
+            {
+                yield return new object[] { "1:1:1.00000001", CultureInfo.InvariantCulture, new TimeSpan(36610000001) };
+            }
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply src/System.Runtime/tests/System/TimeSpanTests.cs]: We'll not fail the parsing on netcoreapp but we'll produce a different correct result. I already covering this case in the added tests to netcoreapp 

```C#
yield return new object[] {"00:00:00.00000001",   new TimeSpan(0)};
```

[the remark this answers is not in the document]
Comment (stephentoub) [reply src/System.Runtime/tests/System/TimeSpanTests.cs]: Ok. Thanks.

File: src/System.Runtime/tests/System/TimeSpanTests.cs [hunk 2]
```diff
@@ -538,7 +538,11 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
             yield return new object[] { "1:1:1.00001", CultureInfo.InvariantCulture, new TimeSpan(36610000100) };
             yield return new object[] { "1:1:1.000001", CultureInfo.InvariantCulture, new TimeSpan(36610000010) };
             yield return new object[] { "1:1:1.0000001", CultureInfo.InvariantCulture, new TimeSpan(36610000001) };
-            yield return new object[] { "1:1:1.00000001", CultureInfo.InvariantCulture, new TimeSpan(36610000001) };
+
+            if (PlatformDetection.IsFullFramework)
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/TimeSpanTests.cs]: Could you add a comment about why these are netfx only?

File: src/System.Runtime/tests/System/TimeSpanTests.netcoreapp.cs [hunk 3]
```diff
@@ -22,6 +22,27 @@ private static IEnumerable<object[]> MultiplicationTestData()
             yield return new object[] {TimeSpan.MaxValue, 0.5, TimeSpan.FromTicks((long)(long.MaxValue * 0.5))};
         }
 
+        private static IEnumerable<object[]> ParseSpecialCasesData()
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/TimeSpanTests.netcoreapp.cs]: Is there any more descriptive naming we could use about what makes these special?

File: src/System.Runtime/tests/System/TimeSpanTests.netcoreapp.cs [hunk 4]
```diff
@@ -22,6 +22,27 @@ private static IEnumerable<object[]> MultiplicationTestData()
...
+            yield return new object[] {"0:00:00.00123456",    new TimeSpan(12346)};
+            yield return new object[] {"0:00:00.00000098",    new TimeSpan(10)};
+            yield return new object[] {"0:00:00.00000099",    new TimeSpan(10)};
+        }
+
+        [Theory, MemberData(nameof(ParseSpecialCasesData))]
+        public static void Multiplication(string input, TimeSpan expected)
+        {
+            Assert.Equal(expected, TimeSpan.Parse(input, CultureInfo.InvariantCulture));
+        }
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/TimeSpanTests.netcoreapp.cs]: Is there benefit to testing ParseExact with these, too?

File: src/System.Runtime/tests/System/TimeSpanTests.netcoreapp.cs (hunk 3 above)
Comment (tarekgh) [reply src/System.Runtime/tests/System/TimeSpanTests.netcoreapp.cs]: I changed the name and added a comment to tell more what we are trying to do. thanks.

**Comment by stephentoub** [comment]
Cherry-picked into https://github.com/dotnet/corefx/pull/34596.

**Comment by tarekgh** [comment]
thanks @stephentoub

**Pull request closed without merging**